api/rosetta/data_integration_test.go

```diff
@@ -103,16 +122,32 @@ func TestGetBalance(t *testing.T) {
 			wantStatusCode: http.StatusUnprocessableEntity,
 			wantHandlerErr: assert.Error,
 		},
+		/*
+			TODO: when we re-add block validation, we should reintroduce this test case - block height and hash mismatch should result in an error
+				=> https://github.com/optakt/flow-dps/issues/51
+			{
+				name:           "invalid request - block hash and height mismatch",
+				request:        balanceRequest("754aed9de6197641", 50, "594d59b2e61bb18b149ffaac2b27b0efe1854f6795cd3bb96a443c3676d78683"),
+				wantStatusCode: http.StatusUnprocessableEntity,
+				wantHandlerErr: assert.Error,
+			},
+		*/
 		{
-			name:           "block hash and height mismatch",
-			request:        getBalanceRequest("8c5303eaa26202d6", 99, "d47b1bf7f37e192cf83d2bee3f6332b0d9b15c0aa7660d1e5322ea964667b333"),
+			name:           "invalid account address",
+			request:        balanceRequest("invalid_address", 0, "d47b1bf7f37e192cf83d2bee3f6332b0d9b15c0aa7660d1e5322ea964667b333"),
 			wantStatusCode: http.StatusUnprocessableEntity,
 			wantHandlerErr: assert.Error,
 		},
 		{
-			name:           "invalid account address",
-			request:        getBalanceRequest("invalid_address", 0, "d47b1bf7f37e192cf83d2bee3f6332b0d9b15c0aa7660d1e5322ea964667b333"),
-			wantStatusCode: http.StatusUnprocessableEntity,
+			name:           "invalid balance request - account does not exist yet",
+			request:        balanceRequest("754aed9de6197641", 12, "9035c558379b208eba11130c928537fe50ad93cdee314980fccb695aa31df7fc"),
+			wantStatusCode: http.StatusInternalServerError,
```

Comment:
Why would it be an internal server error? This should be a bad request in my book 🤔 The request asks for something that does not exist, it's a 404 at best